### PR TITLE
Improve AOT Compilation Accuracy.

### PR DESCRIPTION
**Pull request by apghml**

### CODE REVIEW

File: axlearn/common/launch.py [hunk 1]
```diff
@@ -6,35 +6,17 @@
...
-        # Per rwitten@google.com the following two flags allow gradient all-reduce to happen
-        # concurrently with gradient computation for the following layer.
-        "--xla_tpu_enable_data_parallel_all_reduce_opt=true",
-        "--xla_tpu_data_parallel_opt_different_sized_ops=true",
-    ]
-
-os.environ["LIBTPU_INIT_ARGS"] = " ".join(libtpu_init_args)
+# Set LIBTPU_INIT_ARGS before importing jax!
+libtpu_init_options = compiler_options.default_xla_options(
+    instance_type=instance_type, num_slices=num_tpu_slices, backend="tpu"
```
Comment (markblee) [inline axlearn/common/launch.py:17]: How come `infer_tpu_type` (called from `default_xla_options`) doesn't raise when `instance_type="gpu"`? LMK what I missed.

Comment (apghml) [reply axlearn/common/launch.py]: Ah, I totally misunderstood your comment. Fixed by only calling it if the instance type starts with tpu.

Comment (apghml) [reply axlearn/common/launch.py]: The latest version has also been testing on GPU now.

**Comment by apghml** [comment]
@markblee I think I need your approval again before I can merge.

**Review by markblee — APPROVED** [review]

**Pull request merged**